packages/inspector/src/components/CleanAssets/utils.ts

```diff
@@ -88,15 +108,58 @@ export function collectUsedAssets(sdk: SdkContextValue): Set<string> {
   return usedAssets;
 }
 
+/**
+ * Extracts all external resource paths referenced by a glTF model
+ * @param gltf The parsed glTF data
+ * @returns List of normalized asset paths referenced by the model
+ */
+function extractModelReferencedAssets(gltf: Gltf): string[] {
+  if (!Array.isArray(gltf.info?.resources)) return [];
+
+  // Extract external resources paths (buffers, images, etc.)
+  return gltf.info.resources
+    .filter(resource => resource.storage === 'external' && resource.uri)
+    .map(resource => resource.uri.toLowerCase());
+}
+
 /**
  * Scans the project and identifies unused assets
  * @param sdk The SDK context
  * @param allFiles The list of all asset files with their sizes
  * @returns Array of AssetFile objects with unused flag set
  */
-export function scanForUnusedAssets(sdk: SdkContextValue, allFiles: FileSize[]): AssetFile[] {
+export async function scanForUnusedAssets(
+  sdk: SdkContextValue,
+  allFiles: FileSize[],
+  dataLayer: IDataLayer,
+): Promise<AssetFile[]> {
   const usedAssets = collectUsedAssets(sdk); // Get all assets referenced in the scene
 
+  // Analyze used model assets to find referenced external resources
+  const promises = Array.from(usedAssets.values()).map(async (filePath: string) => {
+    if (isModelAsset(filePath)) {
+      try {
+        if (!dataLayer) return;
+        const fileResponse = await dataLayer.getFile({ path: filePath });
```

Comment:
maybe a new rpc method that receives `path[]` and returns `buffer[]` to avoid doing n rpc calls?